main.go

```diff
@@ -68,10 +69,19 @@ func main() {
 		os.Exit(1)
 	}
 
+	c, err := utils.ReadConfiguration()
+	if err != nil {
+		setupLog.Error(err, "unable to read configuration")
+		os.Exit(1)
+	}
+
+	acdAppClient := utils.GetArgoCDAppClient(c)
+
 	if err = (&controllers.ProgressiveRolloutReconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("ProgressiveRollout"),
-		Scheme: mgr.GetScheme(),
+		Client:          mgr.GetClient(),
+		Log:             ctrl.Log.WithName("controllers").WithName("ProgressiveRollout"),
+		Scheme:          mgr.GetScheme(),
+		ArgoCDAppClient: acdAppClient,
```

Comment:
```suggestion
		ArgoCDAppClient: utils.GetArgoCDAppClient(c),
```

Why not using the factory function in the reconciler definition, as you did with the other lines?